lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -91,14 +104,52 @@ public byte[] Call(byte[] memento)
              
                 _dataReducer.Send(result);
             }
+            Interlocked.Exchange(ref _stopped, 1);
             return null;
         }
 
         /// <summary>
+        /// Received close event. 
+        /// If it is closed event sent from driver, set _break to true so that to stop the iteration in Call method
+        /// After _enforceCloseTimeout, if the task is still not closed, throw IMRUTaskSystemException to enforce stop. 
+        /// </summary>
+        /// <param name="closeEvent"></param>
+        public void OnNext(ICloseEvent closeEvent)
+        {
+            if (closeEvent.Value.IsPresent() && Encoding.UTF8.GetString(closeEvent.Value.Value).Equals(TaskManager.CloseTaskByDriver))
```

Comment:
`Encoding.UTF8.GetString(closeEvent.Value.Value)` is computed twice here. Maybe use a local variable for it?